### PR TITLE
Avoid possibly overflowing hosts.

### PR DESCRIPTION
**Pull request by script3r**
A few more details were sent to `opencomdb2@bloomberg.net`.

### CODE REVIEW

File: cdb2api/cdb2api.c [hunk 1]
```diff
@@ -4763,7 +4763,7 @@ int cdb2_open(cdb2_hndl_tp **handle, const char *dbname, const char *type,
         char *p;
         p = strchr(type, ':');
         if (p) {
-            strcpy(hndl->hosts[0], type);
+            strncpy(hndl->hosts[0], type, sizeof(hndl->hosts[0]));
```
Comment (adizaimi) [inline cdb2api/cdb2api.c]: sizeof - 1?

Comment (script3r) [inline cdb2api/cdb2api.c]: @adizaimi, Indeed, however we still need to explicitly nul-terminate: something like `strlcpy` will get us both in one call.

We can probably perform better validation elsewhere, but I'm still familiarizing myself with the code. I'll patch this with explicit nul termination.

Comment (riverszhang89) [inline cdb2api/cdb2api.c]: `hndl` is allocated via `calloc` so it is fine to leave `NUL`. However I'd agree that explicit nul termination definitely improves the readability.

Comment (akshatsikarwar) [inline cdb2api/cdb2api.c]: Also see `bb/str0.c` for related functions

Comment (script3r) [inline cdb2api/cdb2api.c]: @akshatsikarwar I'll take a look at bb/str0.c, I was actually looking for `misc` string functions.
@riverszhang89, @adizaimi  you are right, the -1 will suffice.

File: cdb2api/cdb2api.c [hunk 2]
```diff
@@ -4763,7 +4763,7 @@ int cdb2_open(cdb2_hndl_tp **handle, const char *dbname, const char *type,
         char *p;
         p = strchr(type, ':');
         if (p) {
-            strncpy(hndl->hosts[0], type, sizeof(hndl->hosts[0]));
+            strncpy(hndl->hosts[0], type, sizeof(hndl->hosts[0])-1);
```
Comment (riverszhang89) [inline cdb2api/cdb2api.c]: I do think our coding style requires spaces around binary operands.

**Comment by script3r** [comment]
There's a bit more, unfortunately, we can still overflow since the following two lines assume that `:` exists in the host, and proceeds to access  the memory; however, since we (possibly) trimmed, we need to account for `:` not being present.

```
p = strchr(hndl->hosts[0], ':');
*p = 0;
```

**Comment by riverszhang89** [comment]
@script3r Actually if `:` is not found, `strchr` returns `NULL`, and we end up dereferencing `NULL`.

**Pull request merged**